braindecode/datasets/base.py

```diff
@@ -194,7 +209,21 @@ def __getitem__(self, index):
         np.ndarray
             Crop indices.
         """
-        X = self.windows.get_data(item=index)[0].astype('float32')
+
+        # necessary to cast as list to get list of three tensors from batch,
+        # otherwise get single 2d-tensor...
+        crop_inds = self.crop_inds[index].tolist()
+
+        i_window_in_trial, i_start, i_stop = crop_inds
+        if hasattr(self.windows, 'metadata'):  # mne epochs case:
+            X = self.windows.get_data(item=index)[0].astype('float32')
+        else:
+            X = self.windows._getitem((slice(None), slice(i_start, i_stop)), return_times=False)
+        X = X.astype('float32')
+        # ensure we don't give the user the option
+        # to accidentally modify the underlying array
+        X = X.copy()
```

Comment:
how much can this slow down `__getitem__`? This will be called all the time during training, curious whether it could be a potential speed up?

Reply:
I would say this is not a big issue as this is done asynchronously by the data loader.
A potential solution would be to use `X.flags.writeable = False` to make the array not writable without copy but this might break a lot of code so I would keep it that way.

Reply:
yeah probably copying single window should be negligible overall...